File: Aptech/C language/C/C/Own try/function_calculator.cpp
#include <stdio.h>
#include <conio.h>
#include <math.h>

// calling functions
float addition(float num1, float num2);
float substraction(float num1, float num2);
float multiplication(float num1, float num2);
float divition(float num1, float num2);
// int reminder(int num1, int num2);
int main()
{
    // Code
    float number_1, number_2, answer;
    char choose;

    while (1)
    {
        /* code */

        // First Number input
        printf("Enter your first Number : \n");
        scanf("%f", &number_1);

        // second number input
        printf("Enter your Second Number : \n");
        scanf("%f", &number_2);

        // Choosing user to which calculation
        printf("What you want to do ? \n");
        printf("Press A for Addition \n");
        printf("Press S for Substraction \n");
        printf("Press M for Multiplication \n");
        printf("Press D for Divition \n");
        // printf("Press R for Reminder \n");

        choose = getche();
        if (choose == 'A' || choose == 'a')
        { /* Addition */
            // answer = addition(number_1, number_2);
            printf("Your answer is : %.2f\n", addition(number_1, number_2));
        }
        else if (choose == 'S' || choose == 's')
        {
            /*Substraction*/
            printf("Your answer is : %.2f\n", substraction(number_1, number_2));
        }
        else if (choose == 'M' || choose == 'm')
        {
            /* Multiplication */
            printf("Your answer is : %.2f\n", multiplication(number_1, number_2));
        }
        else if (choose == 'D' || choose == 'd')
        {
            /* Divition */
            printf("Your answer is : %.2f\n", divition(number_1, number_2));
        }
        // else if (choose == 'R' || choose == 'r')
        // {
        //     /* Reminder */
        //     printf("Your answer is : %.2f", reminder(number_1, number_2));
        // }
        else if (choose == 'e' || choose == 'E')
        {
            /* code */
            exit(0);
        }

        else
        {
            printf("Sorry you have entered %c something wrong \n", choose);
        }
    }

    // printf("\n Your answer is : %.2f", answer);

    getch();
    return 0;
}

// Declating Addition
float addition(float num1, float num2)
{
    return num1 + num2;
}

// Declaring Substraction
float substraction(float num1, float num2)
{
    return num1 - num2;
}

// Declaring multiplication
float multiplication(float num1, float num2)
{
    return num1 * num2;
}

// Declaring divition
float divition(float num1, float num2)
{
    return num1 / num2;
}

// Declaring reminder
// int reminder(int num1, int num2)
// {
//     return num1 % num2;
// }

File: JAVA/Example_java/function_calculator.java
import java.util.Scanner;

//main class
public class function_calculator {

  //addition function
  public static void addition() {
    int fn, sn, answer;
    Scanner input = new Scanner(System.in);
    System.out.println("Java addition");
    System.out.println("Enter first Number :");
    fn = input.nextInt();
    System.out.println("Enter second Number :");
    sn = input.nextInt();
    input.close();
    answer = (fn + sn);
    System.out.println(
      "The addition of " + fn + " and " + sn + " is = " + answer
    );
  }

  //substraction function
  public static void substraction() {
    int fn, sn, answer;
    Scanner input = new Scanner(System.in);
    System.out.println("Java Substraction");
    System.out.println("Enter first Number :");
    fn = input.nextInt();
    System.out.println("Enter second Number :");
    sn = input.nextInt();
    input.close();
    answer = (fn - sn);
    System.out.println(
      "The Substraction of " + fn + " and " + sn + " is = " + answer
    );
  }

  //division function
  public static void division() {
    int fn, sn, answer;
    Scanner input = new Scanner(System.in);
    System.out.println("Java division");
    System.out.println("Enter first Number :");
    fn = input.nextInt();
    System.out.println("Enter second Number :");
    sn = input.nextInt();
    input.close();
    answer = (fn / sn);
    System.out.println(
      "The division of " + fn + " and " + sn + " is = " + answer
    );
  }

  //multiplication function
  public static void multiplication() {
    int fn, sn, answer;
    Scanner input = new Scanner(System.in);
    System.out.println("Java multiplication");
    System.out.println("Enter first Number :");
    fn = input.nextInt();
    System.out.println("Enter second Number :");
    sn = input.nextInt();
    input.close();
    answer = (fn * sn);
    System.out.println(
      "The multiplication of " + fn + " and " + sn + " is = " + answer
    );
  }

  //reminder function
  public static void reminder() {
    int fn, sn, answer;
    Scanner input = new Scanner(System.in);
    System.out.println("Java reminder");
    System.out.println("Enter first Number :");
    fn = input.nextInt();
    System.out.println("Enter second Number :");
    sn = input.nextInt();
    input.close();
    answer = (fn % sn);
    System.out.println(
      "The reminder of " + fn + " and " + sn + " is = " + answer
    );
  }

  //This function will decide to call that calculation function which use input
  public static void calculation() {
    System.out.println("My first java simple calculator");
    System.out.println(
      "Enter what you want to do \n 1. Addition \n 2. Subsraction \n 3. Multiplication \n 4. Division \n 5. Reminder \n ->"
    );
    Scanner input = new Scanner(System.in);
    int decide = input.nextInt();

    if (decide == 1) {
      addition();
    } else if (decide == 2) {
      substraction();
    } else if (decide == 3) {
      multiplication();
    } else if (decide == 4) {
      division();
    } else if (decide == 5) {
      reminder();
    } else {
      System.out.println("Sorry you have entered wrong input " + decide);
    }
    input.close();
  }

  public static void main(String[] args) {
    calculation();
    // int choose;
    // Scanner sc = new Scanner(System.in);
    // do {
    //   calculation();
    //   System.out.println("Do you want to Re Calculate \n 1. YES 2. NO");
    //   choose = sc.nextInt();
    // } while (choose == 1);
    // sc.close();
    // System.out.println("Terminal Quited");
  }
}
